src/MonitoringFunctions/DataService/Kusto/DirectJsonMappingResolver.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Kusto.Data.Common;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MonitoringFunctions.DataService.Kusto
+{
+    /// <summary>
+    /// Creates column mapping for objects whose json property names directly match the column name in the corresponding Kusto table.
+    /// </summary>
+    internal sealed class DirectJsonMappingResolver : IJsonColumnMappingResolver
+    {
+        public IEnumerable<ColumnMapping> GetColumnMappings<T>() where T : IKustoTableRow
+        {
+            DefaultContractResolver contractResolver = new DefaultContractResolver();
+            var contract = contractResolver.ResolveContract(typeof(T)) as JsonObjectContract;
```

Comment:
resolution up to you on this one.

Reply:
`typeof(T)` is used twice: one will execute always, the other one when throwing an exception. The latter is expected to run 1% of the time, maybe less. I think it may even hurt the performance to store the value of the type for a code path that will not execute for 99% of the time.